virtualfish/virtual.fish

```diff
@@ -121,6 +121,46 @@ function __vf_deactivate --description "Deactivate this virtualenv"
     set -e VIRTUAL_ENV
 end
 
+function __vfsupport_find_python --description "Search for and return Python path"
+    set -l python_arg $argv[1]
+    set -l py_path (which $python_arg)
+    set -l py_version (string replace "python" "" $python_arg)
+    set -l brew_path "/usr/local/opt/python@$py_version/bin/python$py_version"
+    # Executable on PATH (python3/python3.8) or full interpreter path
+    if test -x "$py_path"
+        set python "$py_path"
```

Comment:
From the diff I'm looking at, it doesn't appear like you've defined $python anywhere.

That means if there is a global by that name (not too unlikely!) it would be overwritten.

I suggest adding a `set -l python` at the beginning of the function (yeah, block scope really bites us in the butt here)